### DISCUSSION TITLE
Support ssh-keys to propagate build status to Bitbucket

### DISCUSSION THREAD

**Discussion by casareafer2908** [jenkinsci/bitbucket-push-and-pull-request-plugin#212] [Ideas]
It would be simpler to set up Jenkins if the bitbucket-push-and-pull-request-plugin could support ssh keys to propagate the build status to bitbucket

As mentioned by @cdelmonte-zg in #108 , There is a limit while propagating the build status to bitbucket, propagation will work only if you are using, for the git plugin, credentials of the kind: "Username with Password".
https://github.com/jenkinsci/bitbucket-push-and-pull-request-plugin/issues/108#issuecomment-646781172

I will research for other options, I just mention this as a "nice to have" feature.

**Comment by macghriogair** [comment]
This is actually a limitation of how the REST Api authentication with Bitbucket Server and Cloud works. I don't think this plugin can do anything about it.

Reference: 
* https://developer.atlassian.com/server/bitbucket/how-tos/example-basic-authentication/
* https://developer.atlassian.com/bitbucket/api/2/reference/meta/authentication

[a comment by an automated account is not shown]

**Comment by derekatkins** [comment]
@macghriogair --
> This is actually a limitation of how the REST Api authentication with Bitbucket Server and Cloud works. I don't think this plugin can do anything about it.
> 
> Reference:
> 
>     * https://developer.atlassian.com/server/bitbucket/how-tos/example-basic-authentication/
> 
>     * https://developer.atlassian.com/bitbucket/api/2/reference/meta/authentication

Indeed, the Rest API cannot use SSH credentials, however it would be useful if this plug-in could use OAuth Credentials (ala the bitbucket build status notifier plugin).

**Comment by fabiocody** [comment]
Now that Bitbucket has disabled username/password login, how can I propagate the build status back to Bitbucket?
